### PR TITLE
allowed objective to skip some datasets

### PR DESCRIPTION
**Pull request by zaccharieramzi**
This solves #321 

However, I don't really know how I should test this. I understand there is a dummy benchmark, but should I add a case where I skip a dataset i the objective? Like create a fake dataset and always skip it?

### CODE REVIEW

File: benchopt/tests/test_benchmarks/dummy_benchmark/objective.py [hunk 1]
```diff
@@ -16,6 +20,11 @@ def set_data(self, X, y):
         self.X, self.y = X, y
         self.lmbd = self.reg * self._get_lambda_max()
 
+    def skip(self):
```
Comment (tomMoral) [inline benchopt/tests/test_benchmarks/dummy_benchmark/objective.py:23]:
```suggestion
    def skip(self, X, y):
```

**Review by tomMoral — APPROVED** [review]
I would change a bit the API to match the one from `Solver.skip`.
Otherwise, LGTM!, thx @zaccharieramzi

**Comment by zaccharieramzi** [comment]
Yup makes total sense, just added in your suggestions @tomMoral

File: benchopt/tests/test_benchmarks/dummy_benchmark/objective.py [hunk 2]
```diff
@@ -16,6 +20,11 @@ def set_data(self, X, y):
         self.X, self.y = X, y
         self.lmbd = self.reg * self._get_lambda_max()
 
+    def skip(self, X, y):
+        if np.all(self.X == 0):
```
Comment (tomMoral) [inline benchopt/tests/test_benchmarks/dummy_benchmark/objective.py:24]: whoops, i forgot one!

```suggestion
        if np.all(X == 0):
```

**Pull request merged**